tests/suites/hooks/reboot.sh

```diff
@@ -26,6 +26,7 @@ run_start_hook_fires_after_reboot() {
 
     # Restart the unit agent and ensure that the implicit start hook still
     # does not fire
+    wait_for_systemd_service_files_to_appear "ubuntu-lite/0"
```

Comment:
This was causing issues on (slow) CI as there was a small race between the unit appearing as idle and the systemd unit files being written to disk. This change addresses the issue and allows us to ensure that the start hook does not fire when the agent gets restarted _externally_.